### ISSUE TITLE
Instructor: Edit question: Check for empty string in Numscale/MSQ/Rank (Options/Recipients) number fields

### ISSUE THREAD

**Issue by sukanta-27** [TEAMMATES/teammates#8683]
Noticed while working on #8611 

When editing existing questions for rank options/recipient, MSQ, Numscale questions, if we leave a number field blank upon editing and delete the question, then instead of the question getting deleted, a error page in shown.

- **Environment**: 
`master` branch

**Steps to reproduce:**
- Login as instructor
- Edit a existing session
- Edit a existing question from the type mentioned above (or, add a new question, then edit it)
- Edit a number field (e.g. - in the `Numscale` question, edit the max/min/step value and leave it blank)
- `Delete` the question (Don't save the changes before deleting)

**Expected behaviour:** Following the example of question title and other fields, the question should get deleted.

**Actual behaviour:** An error page is shown.

**Additional info:** This happens because of `NumberFormatException`, for passing an empty string (`""`) in `Integer.parseInt(...)`
Now e.g in the `Numscale` question for minimum value we have:
https://github.com/TEAMMATES/teammates/blob/8a974ee703c46f4b3280552f46b3e767eb561eab/src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java#L50-L54

Here we check if the `minScaleString` is null or not, but that is not sufficient. If we are editing a question, then the value of the variable may not be null (As it is pointing towards something), but still can be an empty string, which is why it is passing the assert and passing an empty string into `Integer.parseInt(...)`, causing this error.

**Possible solution:** We can check if the String length is > 0 or not. For the above example,

```java
String minScaleString = 
        HttpRequestHelper.getValueFromParamMap(requestParameters, 
                                               Const.ParamsNames.FEEDBACK_QUESTION_NUMSCALE_MIN); 
Assumption.assertNotNull("Null minimum scale", minScaleString); 
if (minScaleString.length() > 0) {
    int minScale = Integer.parseInt(minScaleString);
}
```

**Comment by bqnguyen94** [comment]
I'm interested in why the `delete question` action requires valid `FEEDBACK_QUESTION_NUMSCALE_MIN` in the first place. Discussion with @LiHaoTan and @Crphang reveals that this is because `edit` and `delete` actions are coupled:
https://github.com/TEAMMATES/teammates/blob/11eb4f64053148a73603fe6eed7b171673d837c3/src/main/java/teammates/ui/controller/InstructorFeedbackQuestionEditAction.java#L47-L65
Looking further into `deleteQuestion(updatedQuestion)`:
https://github.com/TEAMMATES/teammates/blob/11eb4f64053148a73603fe6eed7b171673d837c3/src/main/java/teammates/ui/controller/InstructorFeedbackQuestionEditAction.java#L71-L77
Seems like the only parameters used here are `feedbackQuestionId`, `questionNumber`, `feedbackSessionName`, `courseId`, which can be got without creating a `FeedbackQuestionDetails` as `updatedQuestion`, which is what is throwing the `NumberFormatException` in the first place. Personally, I think the `delete` action should not have to care about the user's unsaved input in the first place.

TL;DR: Another way to solve the problem is to change `deleteQuestion(updatedQuestion)` such that it doesn't need the whole `updatedQuestion`.
